packages/front-end/components/Features/FeatureDiagnostics.tsx

```diff
@@ -0,0 +1,314 @@
+import { FeatureInterface } from "back-end/types/feature";
+import { useState, useMemo } from "react";
+import { useForm } from "react-hook-form";
+import { OrganizationSettings } from "back-end/types/organization";
+import { DataSourceInterfaceWithParams } from "back-end/types/datasource";
+import { isProjectListValidForProject } from "shared/util";
+import { FeatureEvalDiagnosticsQueryResponseRows } from "shared/types/integrations";
+import { QueryStatistics } from "back-end/types/query";
+import { Box, Flex } from "@radix-ui/themes";
+import { getValidDate } from "shared/dates";
+import { format } from "date-fns";
+import Button from "@/ui/Button";
+import { useDefinitions } from "@/services/DefinitionsContext";
+import SelectField from "@/components/Forms/SelectField";
+import useOrgSettings from "@/hooks/useOrgSettings";
+import { useAuth } from "@/services/auth";
+import LinkButton from "@/ui/LinkButton";
+import { useAddComputedFields, useSearch } from "@/services/search";
+import Callout from "@/ui/Callout";
+import Frame from "@/ui/Frame";
+import Link from "@/ui/Link";
+import Tooltip from "@/ui/Tooltip";
+import EmptyState from "@/components/EmptyState";
+
+type FeatureEvaluationDiagnosticsQueryResults = {
+  rows?: FeatureEvalDiagnosticsQueryResponseRows;
+  statistics?: QueryStatistics;
+  error?: string;
+  sql?: string;
+  truncated?: boolean;
+};
+
+// Helper function to format a value for display
+const formatDisplayValue = (value: unknown): string => {
+  if (value === null) {
+    return "";
+  } else if (value === undefined) {
+    return "";
+  } else if (typeof value === "boolean") {
+    return String(value);
+  } else if (typeof value === "object") {
+    return JSON.stringify(value);
+  } else {
+    return String(value);
+  }
+};
+
+export function getFeatureDatasourceDefaults(
```

Comment:
question: Is this function exported? If yes, should we move it to a different helper file? Otherwise we should remove the `export`.

Reply:
nit: Also should we rename it to `getDataSourceInitialFormValue`? or something to add `Form`.

I thought this was related to datasource configuration defaults at first.